File: ISL.Reidentification.WebApps.Configurations/ISL.Reidentification.Configurations.Server.Tests.Acceptance/Models/Lookups/UserAccess.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;

namespace ISL.ReIdentification.Configurations.Server.Tests.Acceptance.Models.Lookups
{
    public class UserAccess
    {
        public Guid Id { get; set; }
        public string UserEmail { get; set; }
        public string RecipientEmail { get; set; }
        public string OrgCode { get; set; }
        public DateTimeOffset ActiveFrom { get; set; }
        public DateTimeOffset? ActiveTo { get; set; }
        public string CreatedBy { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
        public string UpdatedBy { get; set; }
        public DateTimeOffset UpdatedDate { get; set; }
    }
}

File: ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Client/src/models/userAccess/userAccess.ts
export class UserAccess {
    public id: string;
    public firstName: string;
    public lastName: string;
    public userEmail: string;
    public orgCode: string;
    public activeFrom?: Date;
    public activeTo?: Date;
    public createdBy?: string;
    public createdDate?: Date | undefined;
    public updatedBy?: string;
    public updatedDate?: Date | undefined;

    constructor(user: any) {
        this.id = user.id ? user.id : "";
        this.firstName = user.firstName || "";
        this.lastName = user.lastName || "";
        this.userEmail = user.userEmail || "";
        this.orgCode = user.orgCode || "";
        this.activeFrom = user.activeFrom;
        this.activeTo = user.activeTo;
        this.createdDate = user.createdDate ? new Date(user.createdDate) : undefined;
        this.createdBy = user.createdBy || "";
        this.updatedDate = user.updatedDate ? new Date(user.updatedDate) : undefined;
        this.updatedBy = user.updatedBy || "";
    }
}